### PR TITLE
add purescript-npm attack

### PR DESCRIPTION
**Pull request by dim0x69**

### CODE REVIEW

File: supply-chain-security/compromises/README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ We welcome additions to this catalog by
 
 | Name              | Year               | Type of compromise    | Link        |
 | ----------------- | ------------------ | ------------------    | ----------- |
+| [purescript-npm](2019/purescript-npm.md) | 2019 | Source Code Compromise | [1](https://harry.garrood.me/blog/malicious-code-in-purescript-npm-installer/) |
```
Comment (lumjjb) [inline supply-chain-security/compromises/README.md]: Is there a link that points to an organization blog or news page? Would be preferred to someone's personal blog.

File: supply-chain-security/compromises/2019/purescript-npm.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Malicious code was inserted in dependencies of the purescript npm installer. 
+The code was inserted in the packages load-from-cwd-or-npm and rate-map.
+
+
+## Impact
+Both packages have ~ 1400 downloads weekly.
+
+
+## Type of compromise
+The maintainer at that time claims that the malicious code was published by an attacker
```
Comment (lumjjb) [inline supply-chain-security/compromises/2019/purescript-npm.md]: Can you reword this to follow the wording style of other compromises. "It appears ..." when uncertain

File: supply-chain-security/compromises/2019/purescript-npm.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Malicious code in purescript npm installer dependencies
+Malicious code was inserted in dependencies of the purescript npm installer. 
+The code was inserted in the packages load-from-cwd-or-npm and rate-map.
+
+
+## Impact
+Both packages have ~ 1400 downloads weekly.
```
Comment (lumjjb) [inline supply-chain-security/compromises/2019/purescript-npm.md]: Can we elaborate a bit more on timeframe here? "Malicious packages may have been downloaded X times, over Y time?" (assuming we count from the discovery/rectification date).

**Comment by lumjjb** [comment]
hi @dim0x69, are you actively working on this PR? If not I will mark it as inactive.

**Comment by dim0x69** [comment]
Hi @lumjjb  I just updated the PR :)

**Comment by lumjjb** [comment]
Thanks @dim0x69! Looks like there are conflicts with upstream, could you rebase/resolve conflicts? Thanks

@SantiagoTorres can you help look through this?

**Comment by SantiagoTorres** [comment]
@lumjjb sure! let me give it a go now. @dim0x69 if youd like, I can take care of the conflicts myself as well.
